src/code.jl

```diff
@@ -696,6 +697,46 @@ end
 
 @inline newsym(::Type{T}) where T = Sym{T}(gensym("cse"))
 
+"""
+$(SIGNATURES)
+
+Perform a topological sort on a symbolic expression represented as a Directed Acyclic 
+Graph (DAG).
+
+This function takes a symbolic expression `graph` (potentially containing shared common 
+sub-expressions) and returns an array of `Assignment` objects.  Each `Assignment` 
+represents a node in the sorted order, assigning a fresh symbol to its corresponding 
+expression. The order ensures that all dependencies of a node appear before the node itself 
+in the array.
+
+Hash consing is assumed, meaning that structurally identical expressions are represented by 
+the same object in memory. This allows for efficient equality checks using `IdDict`.
+"""
+function topological_sort(graph)
+    sorted_nodes = Assignment[]
+    visited = IdDict()
+
+    function dfs(node)
+        if haskey(visited, node)
+            return visited[node]
+        end
+        if iscall(node)
+            args = map(dfs, arguments(node))
+            new_node = maketerm(typeof(node), operation(node), args, metadata(node))
+            sym = newsym(symtype(new_node))
+            push!(sorted_nodes, sym ← new_node)
+            visited[node] = sym
+            return sym
+        else
+            visited[node] = node
```

Comment:
```suggestion
            return sym
        elseif _is_array_of_symbolics(node)
            new_node = map(dfs, node)
            sym = newsym(typeof(new_node))
            push!(sorted_nodes, sym ← new_node)
            visited[node] = sym
            return sym
        else
            visited[node] = node
```

This enables it to also handle arrays of symbolics